js/path.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+export function pathBackwards(path: string): string {
+  path = path.trim();
+  path = path.replace(/\//g, "\\");
+  const double = /\\\\/;
+  while (double.test(path)) {
+    path = path.replace(double, "\\");
+  }
+  return path;
+}
+
+export function pathForwards(path: string): string {
+  path = path.trim();
```

Comment:
I really hate reassignment to arguments, because especially when it is happens with non-primatives, as it is just totally confusing.  Also, the reassignments are a bit confusing.  I would think this would be a lot cleaner:

```ts
let result = path
  .trim()
  .replace(/^([a-zA-Z]+:|\.\/)/, "")
  .replace(/\\/g, "/");
const double = /\/\//;
while (double.test(result)) {
  result = result.replace(result, "/");
}
return result;
```

Also, is there a specific edge case that would not cause the following to work:

```ts
return path
  .trim()
  .replace(/^([a-zA-Z]+:|\.\/)/, "")
  .replace(/\\/g, "/");
  .replace(/\/{2}/g, "/");
```